onnx/inliner/inliner.cc

```diff
@@ -0,0 +1,325 @@
+// Copyright (c) ONNX Project Contributors
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <functional>
+
+#include "onnx/common/assertions.h"
+#include "onnx/inliner/inliner.h"
+
+namespace ONNX_NAMESPACE {
+namespace inliner {
+
+// Attribute lookup function. Returns nullptr if attribute is not found.
+using AttributeLookupFunction = std::function<const AttributeProto*(const std::string& name)>;
+
+using AttributeMap = std::unordered_map<std::string, const AttributeProto*>;
+
+// Function lookup function. Returns true iff lookup is successful, in which case
+// the found FunctionProto is copied into *return_value. The opset version is not
+// a parameter since it is determined by the domain for a given model.
+using FunctionResolver =
+    std::function<bool(const std::string& domain, const std::string& op, FunctionProto* return_value)>;
+
+// We use a string of the form "domain::name" as the id for a function.
+using FunctionId = std::string;
+
+FunctionId GetFunctionId(const FunctionProto& function) {
```

Comment:
Do we want to move the functions etc into an unnamed namespace?

Reply:
I was reading https://google.github.io/googletest/advanced.html#testing-private-code and https://google.github.io/styleguide/cppguide.html#Internal_Linkage

Reply:
Done, thanks!